In [101]:
409 - документов без пинов

count_pins
0              1
1              0
2              7
3              0
4              1
5              0
6              0
7              2
8              0
9              0
10             0
11             5
12             1
13             0
14             3
15             1
16             0
17             2
18             0
19             2
20             0
21             2
22             1
23             0
24             2
25             0
26             1
27             6
28             1
29             3
...          ...
1498           2
1499           4
1500           2
1501           0
1502           2
1503           3
1504           3
1505           0
1506           8
1507           0
1508           2
1509           1
1510           0
1511           2
1512           1
1513           5
1514           0
1515           3
1516           2
1517           9
1518           2
1519          12
1520           0
1521           1
1522           1
1523           1
1524           2
1525           1
1526           2
1527           0

[1528 rows x 1 columns]

### валидация адреса
###### all - выборка; true - адреса совпадают; false - не совпадают; none - не определился  
all - 1528; true - 372; false -1030; none - 126  — если сравнивать с учётом номера дома  
all - 1528; true - 933; false - 469; none - 126  — срванивать только город и улицу  


## распределение количества пинов в документах  

###### без документов в которых 0 пинов (их 409)

In [94]:
import plotly.express as px
# df = px.data.tips()
fig = px.histogram(df[df.count_pins != 0], x="count_pins",  marginal="box",
                   hover_data=df.columns)
fig.show()

## распределение пинов  
### можно выявить (узкие) самые проблемыне моменты

In [100]:
fig = px.histogram(dfp, x="cnt",
                   marginal="box", # or violin, rug
                   hover_data=dfp.columns)
fig.show()

## статистика по укрупнёнынм пинам

In [79]:
fig = px.bar(x=dfi['index'], y=dfi['cnt'], labels={'x':'pin', 'y':'count'})
fig.show()

## ещё один способ смотреть распределение

In [92]:
fig = px.histogram(dfe, x="cnt",
                   marginal="box", nbins=len(dfe)*2,hover_data=dfe.columns)
fig.show()